hack/deployer/cmd/cleanup.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/cloud-on-k8s/v2/hack/deployer/runner"
+)
+
+func CleanupCommand() *cobra.Command {
+	var (
+		errUnimplemented = errors.New("unimplemented")
+		provider         string
+		plansFile        string
+	)
+	var cleanupCmd = &cobra.Command{
+		Use:   "cleanup",
+		Short: "Runs the cleanup operation to cleanup clusters older than 3 days in the given provider.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			switch provider {
+			case runner.GKEDriverID:
+				return cleanup(plansFile, []string{"gke-ci"}, &runner.GKEDriverFactory{})
+			case runner.AKSDriverID:
+				return cleanup(plansFile, []string{"aks-ci"}, &runner.AKSDriverFactory{})
+			case runner.OCPDriverID:
+				return errUnimplemented
+			case runner.EKSDriverID:
+				return cleanup(plansFile, []string{"eks-ci", "eks-arm-ci"}, &runner.EKSDriverFactory{})
+			case runner.KindDriverID:
+				return errUnimplemented
+			default:
+				return fmt.Errorf("unknown provider %s", provider)
+			}
+		},
+	}
+
+	cleanupCmd.Flags().StringVar(&plansFile, "plans-file", "config/plans.yml", "File containing execution plans.")
+	cleanupCmd.Flags().StringVar(&provider, "provider", "gke", "Provider to use.")
+
+	return cleanupCmd
+}
+
+// cleanup will attempt to cleanup any clusters older than 3 days
+func cleanup(plansFile string, planNames []string, driverFactory runner.DriverFactory) error {
+	plans, err := runner.ParsePlans(plansFile)
+	if err != nil {
+		return err
+	}
+	for _, planName := range planNames {
+		var ciPlan *runner.Plan
+		for _, plan := range plans.Plans {
+			if plan.Id == planName {
+				p := plan
+				ciPlan = &p
+			}
+		}
+		if ciPlan == nil {
+			return fmt.Errorf("couldn't default ci plan %s", planName)
+		}
+		client, err := driverFactory.Create(*ciPlan)
+		if err != nil {
+			return err
+		}
+		clusters, err := client.Cleanup()
+		if err != nil {
+			return err
+		}
+		for _, cluster := range clusters {
+			log.Printf("deleted cluster: %s", cluster)
+		}
```

Comment:
Not sure about the interest of this log.More generally, I think we could lighten up what we log a bit because it's quite verbose.

Reply:
I commented the deletion to test. With the deletion on, there is a third log `deleting cluster eck-e2e-stack-8-5-3-cgno-4899`.
```sh
> go run main.go cleanup --provider gke
2023/07/20 13:15:20 Authenticating as service account...
Updated property [core/project].
Activated service account credentials for: [k8s-operator-ci@elastic-cloud-dev.iam.gserviceaccount.com]
eck-e2e-gke-all-gmvc-4912
eck-e2e-stack-8-5-3-cgno-4899
eck-e2e-stack-8-5-3-zbjv-239
2023/07/20 13:15:25 deleting cluster: eck-e2e-gke-all-gmvc-4912
2023/07/20 13:15:25 Deleting cluster...
2023/07/20 13:15:25 deleting cluster: eck-e2e-stack-8-5-3-cgno-4899
2023/07/20 13:15:25 Deleting cluster...
2023/07/20 13:15:25 deleting cluster: eck-e2e-stack-8-5-3-zbjv-239
2023/07/20 13:15:25 Deleting cluster...
2023/07/20 13:15:25 deleted cluster: eck-e2e-gke-all-gmvc-4912
2023/07/20 13:15:25 deleted cluster: eck-e2e-stack-8-5-3-cgno-4899
2023/07/20 13:15:25 deleted cluster: eck-e2e-stack-8-5-3-zbjv-239
```

With `eks`, we also log `null`. I think it's come from the execution of an eks command executed to get the list of clusters to delete.
```
> go run main.go cleanup --provider eks
2023/07/20 13:17:07 dry-run mode enabled
2023/07/20 13:17:15 dry-run mode enabled
eck-e2e-eks-arm-qdnz-5023
null
2023/07/20 13:17:21 deleting cluster eck-e2e-eks-arm-qdnz-5023
2023/07/20 13:17:21 Deleting cluster eck-e2e-eks-arm-qdnz-5023...
2023/07/20 13:17:21 deleted cluster: eck-e2e-eks-arm-qdnz-5023
```

Reply:
I've adjusted to use `WithoutStreaming` to most of these commands, and removed any extraneous print statements.